dipy/align/imaffine.py

```diff
@@ -0,0 +1,715 @@
+            if sigmas is None:
+                sigmas = [3, 1, 0]
+            self.factors = factors
+            self.sigmas = sigmas
+
+    def _init_optimizer(self, static, moving, transform, x0,
+                        static_grid2space, moving_grid2space, prealign):
+        r"""Initializes the registration optimizer
+
+        Initializes the optimizer by computing the scale space of the input
+        images
+
+        Parameters
+        ----------
+        static: array, shape (S, R, C) or (R, C)
+            the image to be used as reference during optimization.
+        moving: array, shape (S, R, C) or (R, C)
+            the image to be used as "moving" during optimization. It is
+            necessary to pre-align the moving image to ensure its domain
+            lies inside the domain of the deformation fields. This is assumed
+            to be accomplished by "pre-aligning" the moving image towards the
+            static using an affine transformation given by the 'prealign'
+            matrix
+        transform: instance of Transform
+            the transformation with respect to whose parameters the gradient
+            must be computed
+        x0: array, shape (n,)
+            parameters from which to start the optimization. If None, the
+            optimization will start at the identity transform. n is the
+            number of parameters of the specified transformation.
+        static_grid2space: array, shape (dim+1, dim+1)
+            the voxel-to-space transformation associated to the static image
+        moving_grid2space: array, shape (dim+1, dim+1)
+            the voxel-to-space transformation associated to the moving image
+        prealign: string, or matrix, or None
+            If string:
+                'mass': align centers of gravity
+                'origins': align physical coordinates of voxel (0,0,0)
+                'centers': align physical coordinates of central voxels
+            If matrix:
+                array, shape (dim+1, dim+1)
+            If None:
+                Start from identity
+        """
+        self.dim = len(static.shape)
+        self.transform = transform
+        self.nparams = transform.get_number_of_parameters()
+
+        if x0 is None:
+            x0 = self.transform.get_identity_parameters()
+        self.x0 = x0
+        if prealign is None:
+            self.prealign = np.eye(self.dim + 1)
+        elif prealign == 'mass':
+            self.prealign = aff_centers_of_mass(static, static_grid2space,
+                                                moving, moving_grid2space)
+        elif prealign == 'origins':
+            self.prealign = aff_origins(static, static_grid2space, moving,
+                                        moving_grid2space)
+        elif prealign == 'centers':
+            self.prealign = aff_geometric_centers(static, static_grid2space,
+                                                  moving, moving_grid2space)
+        else:
```

Comment:
Do you want to check whether this is an array, before setting it? If someone typos one of the other options (e.g. types `masss` instead of `mass`), you probably want to raise an error, instead of silently assigning this string to `self.prealign`